docs/language/functions/grok.md

```diff
@@ -1,38 +1,159 @@
 ### Function
 
-&emsp; **grok** &mdash; parse a string using a grok pattern
+&emsp; **grok** &mdash; parse a string using a Grok pattern
 
 ### Synopsis
 
 ```
-grok(p: string, s: string) -> any
-grok(p: string, s: string, definitions: string) -> any
+grok(p: string, s: string) -> record
+grok(p: string, s: string, definitions: string) -> record
```

Comment:
I'm not sure if the return of `any` that was there before was trying to communicate something subtle that I'm missing. However, since the text below has always said it "returns a record" I went ahead and made this change. But if there's more to the story we can change it back and figure out what else needs to change in the text.

Reply:
`any` is probably more correct since `record` isn't an explicit type, but then again we does use `record` in docs for `nest_dotted` and it does probably make things clearer to the new user so I think this fine.

Reply:
We should add "record" to docs/language/conventions.md.